app/uk/gov/hmrc/tai/service/AuditService.scala

```diff
@@ -64,16 +67,19 @@ trait AuditService {
     ))
   }
 
-  def sendUserEntryAuditEvent(nino: Nino, path: String)(implicit hc: HeaderCarrier): Future[AuditResult] = {
-    taiService.taiSession(nino, TaxYear().year, "") flatMap (sessionData => {
-      val details = Map(
-        "authProviderId" -> authProviderId(hc),
-        "nino" -> nino.nino,
-        "noOfCurrentYearEmployments" -> sessionData.taxSummaryDetailsCY.taxCodeDetails.flatMap(_.employment).getOrElse(Nil).size.toString,
-        "noOfTaxCodes" -> sessionData.taxSummaryDetailsCY.taxCodeDetails.flatMap(_.taxCode).getOrElse(Nil).size.toString
-      )
-      createAndSendAuditEvent(userEnterEvent, path, details)
-    })
+  def sendUserEntryAuditEvent(nino: Nino, path: String, employments: Seq[Employment], taxCodes: TaiResponse)(implicit hc: HeaderCarrier): Future[AuditResult] = {
```

Comment:
Can you change the methos parameters to numberOfEmployments and numberOfTaxCodeIncomes and do the logic in the controller please?